README.md

```diff
@@ -27,7 +23,6 @@ kind: OpenTelemetryCollector
 metadata:
   name: simplest
 spec:
-  image: otel/opentelemetry-collector:latest
```

Comment:
It was required in the past, when the image wasn't available in the public repositories (AFAIR), but it's not required anymore. As such, because it's `simplest`, I decided to remove it.